lib/iris/fileformats/netcdf.py

```diff
@@ -1356,6 +1381,109 @@ def _create_cf_dimensions(
                     size = self._existing_dim[dim_name]
                 self._dataset.createDimension(dim_name, size)
 
+    def _add_mesh(self, cube):
+        """
+        Add the cube's mesh, and all related variables to the dataset.
+        Includes all the mesh-element coordinate and connectivity variables.
+
+        ..note::
+
+            Here, we do *not* add the relevant referencing attributes to the
+            data-variable, because we want to create the data-variable later.
+
+        Args:
+
+        * cube (:class:`iris.cube.Cube`):
+            A :class:`iris.cube.Cube` to be saved to a netCDF file.
+
+        Returns:
+            * cf_mesh_name (string or None):
+            The name of the mesh variable created, or None if the cube does not
+            have a mesh.
+
+        """
+        cf_mesh_name = None
+        mesh = cube.mesh
+        if mesh:
+            if mesh in self._name_coord_map.coords:
+                cf_mesh_name = self._name_coord_map.name(mesh)
+            else:
+                cf_mesh_name = self._create_mesh(cube, mesh)
+                self._name_coord_map.append(cf_mesh_name, mesh)
+
+            cf_mesh_var = self._dataset.variables[cf_mesh_name]
+
+            # Get the mesh-element dim names.
+            mesh_dims = self._mesh_dims[mesh]
+
+            # Add all the element coordinate variables.
+            for location in MESH_LOCATIONS:
```

Comment:
@pp-mo Could you use `mesh.all_coords` instead? Then take advantage of the returned `namedtuple` API?

Reply:
Well, mostly I want to clearly control the order I'm doing them in.
I also still have doubts about the hard-wiring of terms like 'node_x', 'node_y' in the Mesh APIs :  I wish we could be more UGRID-generic.
Which this way --i.e. iterating over the coords of a specific location-- does deliver OK